helpers/classes.py

```diff
@@ -0,0 +1,153 @@
+import re
+from .config import LMS_URL
+
+class Block:
+    '''
+    Represents each block in a course
+    '''
+    def __init__(self, block_dict) -> None:
+        self.usage_key = block_dict.get('id', '')
+        self.id = block_dict.get('block_id', '')
+        self.lms_url = block_dict.get('lms_web_url', '')
+        self.studio_url = None
+        self.type = block_dict.get('type', '')
+        self.display_name = block_dict.get('display_name', '')
+        self.children = block_dict.get('children', None)
+        self.parent = None
+        self.data = None
+
+    def __str__(self) -> str:
+        return f"{self.display_name} : {self.usage_key}"
+    
+    def print_ancestry(self):
+        if self.parent:
+            ancestry = self.parent.print_ancestry()
+            return f"{ancestry} -> {self.usage_key}"
+        
+        return self.usage_key
+    
+    def get_studio_url(self):
+        '''
+        If studio url not present in the leaf block, return the url
+        of the parent block.
+        '''
+        if self.studio_url:
+            return self.studio_url
+        elif self.parent:
+            return self.parent.get_studio_url()
+        else:
+            return None
+        
+    def get_lists_of_links(self):
+        jump_to_list = set()
+        other_course_list = set()
+        external_list = set()
+        if self.data:
+            # Find all strings starting with `href=` and discard the `href="`
+            list_of_links = [link.split('"')[1] for link in re.findall('href=.+?(?=\\">)', self.data)]
+
+            for link in list_of_links:
+                # If link has /jump_to_id/ its an internal link
+                if re.search('\/jump_to_id\/', link):
+                    jump_to_list.add(link.split('/')[2].split('\\')[0])
+
+                # If link has the lms url in it then its a link to other courses
+                elif re.search(LMS_URL, link):
+                    other_course_list.add(link.split('\\')[0])
+
+                # All other links starting with http are external links
+                elif link.startswith('http'):
+                    external_list.add(link.split('\\')[0])
+
+        return (jump_to_list, other_course_list, external_list)
+
+class Course:
+    '''
+    Represents a course
+    '''
+    def __init__(self, course_dict) -> None:
+        self.name = course_dict['name']
+        self.id = course_dict['id']
+        self.url = ''
+        self.blocks_url = course_dict['blocks_url']
+        self.blocks = {}
+        self.block_id_map = {}
+        self.jump_pairs = []
+        self.other_courses = []
+        self.external_links = []
+
+    def __str__(self) -> str:
```

Comment:
```suggestion
    def __repr__(self) -> str:
```

This will also work for debugging.